examples/synchronous-proxy/proxy/communications.rb

```diff
@@ -0,0 +1,122 @@
+module SynchronousProxy
+  module Proxy
+    # We support talking between two workflows using these helper methods. Each workflow
+    # that wants to communicate will include this module. Unlike the Go examples which use
+    # channels, the Ruby support is via #on_signal (to receive), Temporal.signal_workflow
+    # (to send), and #wait_for (to block and wait for incoming signal).
+    #
+    # The basic trick is that we register a single #on_signal signal handler per workflow
+    # via the #setup_signal_handler method. Each incoming signal is parsed to determine if
+    # it's a request or response and then the appropriate ivar is set. After the signal handler
+    # runs, this client executes the block attached to the #wait_for method to see if it
+    # returns true. If the block evaluates that it has received a value into the ivar, it
+    # returns true and unblocks.
+    #
+    module Communications
+      RequestSignalName = "proxy-request-signal".freeze
+      ResponseSignalName = "proxy-response-signal".freeze
+
+      SignalDetails = Struct.new(
+        :name, :key, :value, :error, :calling_workflow_id,
+        keyword_init: true
+      ) do
+        def error?
+          key == "error"
+        end
+
+        def to_input
+          [calling_workflow_id, name, key, value]
+        end
+
+        def self.from_input(input)
+          new({name: input[1], key: input[2], value: input[3], calling_workflow_id: input[0]})
```

Comment:
nit: Ruby 3+ doesn't like it when a hash is used to pass keyword arguments. You either need to use `**{my: 'hash'}` or `my: hash`

Reply:
Will fix. Thanks!